builder/vmware/common/step_vnc_connect.go

```diff
@@ -0,0 +1,137 @@
+package common
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/url"
+
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/mitchellh/go-vnc"
+	"golang.org/x/net/websocket"
+)
+
+type StepVNCConnect struct {
+	VNCEnabled       bool
+	VNCOverWebsocket bool
+	DriverConfig     *DriverConfig
+}
+
+func (s *StepVNCConnect) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	if !s.VNCOverWebsocket && !s.VNCEnabled {
+		return multistep.ActionContinue
+	}
+	ui := state.Get("ui").(packer.Ui)
+
+	var c *vnc.ClientConn
+	var err error
+
+	if s.VNCOverWebsocket {
+		ui.Say("Connecting to VNC over websocket...")
+		c, err = s.ConnectVNCOverWebsocketClient(state)
+	} else {
+		ui.Say("Connecting to VNC...")
+		c, err = s.ConnectVNC(state)
+	}
+
+	if err != nil {
+		state.Put("error", err)
+		ui.Error(err.Error())
+		return multistep.ActionHalt
+	}
+
+	state.Put("vnc_conn", c)
+	return multistep.ActionContinue
+}
+
+func (s *StepVNCConnect) ConnectVNCOverWebsocketClient(state multistep.StateBag) (*vnc.ClientConn, error) {
+	driver := state.Get("driver").(*ESX5Driver)
+
+	// Acquire websocket ticket
+	ticket, err := driver.AcquireVNCOverWebsocketTicket()
+	if err != nil {
+		err := fmt.Errorf("Error acquiring vnc over websocket ticket: %s", err)
+		state.Put("error", err)
+		return nil, err
+	}
+	host := ticket.Host
+	if len(host) == 0 {
```

Comment:
Seems odd to check the host and port values. Have you seen these not be filled in correctly?

Reply:
Yes, with my tests the `ticket.Host` is not filled, is just an empty string. The VM logs report that the host couldn't be found but without any further details of why.